src/commands/react-native/upload.ts

```diff
@@ -0,0 +1,319 @@
+      ],
+    ],
+  })
+
+  private buildVersion?: string
+  private bundle?: string
+  private cliVersion: string
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY,
+    datadogSite: process.env.DATADOG_SITE || 'datadoghq.com',
+  }
+  private disableGit?: boolean
+  private dryRun = false
+  private maxConcurrency = 20
+  private platform?: RNPlatform
+  private projectPath: string = process.cwd() || ''
+  private releaseVersion?: string
+  private repositoryURL?: string
+  private service?: string
+  private sourcemap?: string
+
+  constructor() {
+    super()
+    this.cliVersion = require('../../../package.json').version
+  }
+
+  public async execute() {
+    if (!this.releaseVersion) {
+      this.context.stderr.write('Missing release version\n')
+
+      return 1
+    }
+
+    if (!this.buildVersion) {
+      this.context.stderr.write('Missing build version\n')
+
+      return 1
+    }
+
+    if (!this.service) {
+      this.context.stderr.write('Missing service\n')
+
+      return 1
+    }
+
+    if (!this.bundle) {
+      this.context.stderr.write('Missing bundle path\n')
+
+      return 1
+    }
+
+    if (!this.platform) {
+      this.context.stderr.write('Missing platform\n')
+
+      return 1
+    }
+
+    if (!RN_SUPPORTED_PLATFORMS.includes(this.platform)) {
+      this.context.stderr.write(
+        `Platform ${this.platform} is not supported.\nSupported platforms are ios and android.\n`
+      )
+
+      return 1
+    }
+
+    if (!this.sourcemap) {
+      this.context.stderr.write('Missing sourcemap file path\n')
+
+      return 1
+    }
+
+    this.context.stdout.write(
+      renderCommandInfo(
+        this.bundle,
+        this.sourcemap,
+        this.platform,
+        this.releaseVersion,
+        this.service,
+        this.maxConcurrency,
+        this.dryRun,
+        this.projectPath,
+        this.buildVersion
+      )
+    )
+    const metricsLogger = getMetricsLogger({
+      datadogSite: process.env.DATADOG_SITE,
+      defaultTags: [
+        `version:${this.releaseVersion}`,
+        `build:${this.buildVersion}`,
+        `service:${this.service}`,
+        `cli_version:${this.cliVersion}`,
+        'react-native:true',
+        `platform:${this.platform}`,
+      ],
+      prefix: 'datadog.ci.sourcemaps.upload.',
+    })
+    const apiKeyValidator = newApiKeyValidator({
+      apiKey: this.config.apiKey,
+      datadogSite: this.config.datadogSite,
+      metricsLogger: metricsLogger.logger,
+    })
+    const useGit = this.disableGit === undefined || !this.disableGit
+    const initialTime = Date.now()
+    const payloads = await this.getPayloadsToUpload(useGit)
+    const requestBuilder = this.getRequestBuilder()
+    const uploadMultipart = this.upload(requestBuilder, metricsLogger, apiKeyValidator)
+    try {
+      const results = await asyncPool(this.maxConcurrency, payloads, uploadMultipart)
```

Comment:
I don't see a need for the asyncPool in this command since there is only a single upload.

Reply:
I agree on this. However we are going to support multi-file upload later (in 1-2 months).
I believed it would be easier to keep this code until we need it, rather than temporarily changing it because we don't need such functionality - anyway it was much quicker for me to make that command work.

I don't have a very big opinion on this, I'm ok to make the change as well, I should be able to revert it quickly in a few weeks when we do multi-files upload.